app/jobs/remove_user_event.rb

```diff
@@ -0,0 +1,14 @@
+# Documentation: https://developer.github.com/v3/activity/events/types/#membershipevent
+class RemoveUserJob < ApplicationJob
+  queue_as :github_event
+  def perform(payload)
+    return true unless payload_body.dig('action') == 'removed'
+
+    github_user_id = payload_body.dig('member', 'id')
+    github_organization_id = payload_body.dig('organization','id')
+
+    organization = Organization.find_by(github_id: github_organization_id)
+    organization.users.delete(github_id: github_user_id)
```

Comment:
Instead of deleting this in the job can you mark this for deletion and send it to the `DestroyResourceJob`?

```ruby
user = organization.users.find_by(id: github_user_id)
return true if user.nil?

user.update_attributes(deleted_at: Time.zone.now)
DestroyResourceJob.perform_later(current_organization)
```